frontend/packages/ceph-storage-plugin/src/components/modals/restore-pvc-modal/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,151 @@
+import './_restore-pvc-modal.scss';
+
+import * as React from 'react';
+
+import { Form, FormGroup, Grid, GridItem, TextInput } from '@patternfly/react-core';
+import {
+  HandlePromiseProps,
+  ResourceIcon,
+  withHandlePromise,
+} from '@console/internal/components/utils/index';
+import { K8sResourceKind, k8sCreate, k8sGet } from '@console/internal/module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+import { NamespaceModel, PersistentVolumeClaimModel } from '@console/internal/models';
+import { getName, getNamespace } from '@console/shared';
+
+import { VolumeSnapshotModel } from '../../../models';
+
+export const RestorePVCModal = withHandlePromise((props: RestorePVCModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress, handlePromise } = props;
+  const [submitDisabled, setSubmitDisabled] = React.useState(false);
+  const [pvcResource, setResource] = React.useState(null);
+  const [restorePVCName, setPVCName] = React.useState(`${getName(resource) || 'pvc'}-restore`);
+
+  const onRestoreNameChange = (value: string) => {
+    setSubmitDisabled(!value);
+    setPVCName(value);
+  };
+
+  React.useEffect(() => {
+    k8sGet(PersistentVolumeClaimModel, resource?.spec?.source?.name, getNamespace(resource))
+      .then((obj) => setResource(obj))
+      .catch(() => {
+        setResource(null);
+      });
+  }, [resource]);
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const snapshotName = getName(resource);
+    const pvcSize = resource?.status?.restoreSize;
+    const accessModes = pvcResource?.spec?.accessModes;
+    const pvcStorageClass = pvcResource?.spec?.storageClassName;
+    const namespace = `${getNamespace(resource)}`;
+    const restorePVCTemplate: K8sResourceKind = {
+      apiVersion: PersistentVolumeClaimModel.apiVersion,
+      kind: PersistentVolumeClaimModel.kind,
+      metadata: {
+        name: restorePVCName,
+      },
+      spec: {
+        storageClassName: pvcStorageClass,
+        dataSource: {
+          name: snapshotName,
+          kind: VolumeSnapshotModel.kind,
+          apiGroup: 'snapshot.storage.k8s.io',
+        },
+        accessModes: [...accessModes],
+        resources: {
+          requests: {
+            storage: pvcSize,
+          },
+        },
+      },
+    };
+    handlePromise(k8sCreate(PersistentVolumeClaimModel, restorePVCTemplate, { ns: namespace }))
+      .then(() => {
+        close();
+      })
+      .catch(() => {
+        setSubmitDisabled(true);
```

Comment:
We don't display the error message? How does the user know something went wrong?

Reply:
handlePromise sets the errorMessage.